packages/@glimmer/application/src/application.ts

```diff
@@ -129,22 +150,122 @@ export type Notifier = [() => void, (err: Error) => void];
 const DEFAULT_DOCUMENT = typeof document === 'object' ? document : null;
 
 /**
- * The central control point for starting and running Glimmer components.
+ * A base application class that can be shared across different Glimmer application implementations.
+ * It sets up dependency injection and other configuration.
+ * For example, this base class can be reused across a server-side Glimmer application and a client-side Glimmer application.
  *
- * @public
+ * @internal
  */
-export default class Application implements Owner {
+export abstract class BaseApplication implements Owner {
```

Comment:
Can we break this out into a separate file? We generally try to have a single class (as the default export) per file in this codebase.

Reply:
Yup moved it out and all shared types used across both apps into base app as well.